File: TeamCode/src/main/java/com/info1robotics/bobot/tasks/ClawTask.java
package com.info1robotics.bobot.tasks;

/**
 * Toggles the claw's state.
 */
public class ClawTask extends Task {
    public static double openPosition = 0.5;
    public static double closedPosition = 1.0;

    @Override
    public void run() {
        if (!context.clawOpen) {
            context.claw.setPosition(openPosition);
            context.clawOpen = true;
        }
        else {
            context.claw.setPosition(closedPosition);
            context.clawOpen = false;
        }
        System.out.println(context.clawOpen);
        state = State.FINISHED;
    }
}

File: TeamCode/src/main/kotlin/com/info1robotics/bobot/tasks/ClawTask.kt
package com.info1robotics.bobot.tasks

class ClawTask:Task() {

public  companion object{
    val openPosition=.55
    val closedPosition=.95
}

    override fun tick() {

        state = State.FINISHED;
    }
    override fun run()
    {
        if(!context.clawOpen) {
            context.claw.position=openPosition

        }
        else {
            context.claw.position=closedPosition
        }
        context.clawOpen = !context.clawOpen
    }
}